File: evd_authoring_app/src/model/evdScript/program/skills/closeGripper.js
import { Skill } from '../skill';
import { Gripper } from '../primitives';
import { NodeParser } from '../../utilityFunctions';
import Blockly from 'blockly';


export class CloseGripper extends Skill {

    /*
    * Data structure methods
    */

    static typeString() {
        return 'close-gripper';
    }

    static fullTypeString() {
        return Skill.fullTypeString() + CloseGripper.typeString();
    }

    constructor(thingUuid=null, position=100, effort=100, speed=100, 
                type='', name='', uuid=null, parent=null, appendType=true, 
                primitives=null) 
    {

        if (primitives !== null) {
            primitives = [
                Gripper(
                    thingUuid,
                    position,
                    effort,
                    speed
                )
            ];
        }

        super(
            primitives,
            (appendType) ? 'close-gripper.'+type : type,
            name,
            uuid,
            parent,
            appendType
        );
    }

    static fromDict(dct) {
        return new CloseGripper(
            null,
            100,
            100,
            100,
            dct.type,
            dct.name,
            dct.uuid,
            false,
            dct.primitives.map(p => NodeParser(p))
        );
    }

    static BlocklyToolbox() {
        return { type: 'close_gripper' };
    }

    static BlocklyBlock() {
        return { key: 'close_gripper', data: {
            init: function() {
                this.appendDummyInput()
                    .appendField("Close Gripper");
                this.appendValueInput("thing")
                    .setCheck("thing")
                    .appendField("Thing");
                this.appendDummyInput()
                    .appendField("Position")
                    .appendField(new Blockly.FieldNumber(100, 100, 100), "position")
                    .appendField("%");
                this.appendDummyInput()
                    .appendField("Effort")
                    .appendField(new Blockly.FieldNumber(100, 0, 100), "effort")
                    .appendField("%");
                this.appendDummyInput()
                    .appendField("Speed")
                    .appendField(new Blockly.FieldNumber(100, 0, 100), "speed")
                    .appendField("%");
                this.setInputsInline(false);
                this.setPreviousStatement(true, null);
                this.setNextStatement(true, null);
                this.setColour(210);
                this.setTooltip("close-gripper");
                this.setHelpUrl("close-gripper");
            }
        }};
    }
}

File: evd_ros_backend/evd_ros_core/src/evd_script/program/skills/close_gripper.py
from ..skill import Skill
from ..primitives import Gripper


class CloseGripper(Skill):

    '''
    Data structure methods
    '''

    @classmethod
    def type_string(cls, trailing_delim=True):
        return 'close-gripper' + '.' if trailing_delim else ''

    @classmethod
    def full_type_string(cls):
        return Skill.full_type_string() + cls.type_string()

    def __init__(self, position=100, effort=100, speed=100, thing_uuid=None, type='',
                 name='', uuid=None, parent=None, append_type=True, primitives=None):

        if primitives == None:
            primitives=[
                Gripper(
                    thing_uuid=thing_uuid,
                    position=position,
                    effort=effort,
                    speed=speed)
            ]

        super(CloseGripper,self).__init__(
            type=CloseGripper.type_string() + type if append_type else type,
            name=name,
            uuid=uuid,
            parent=parent,
            append_type=append_type,
            primitives=primitives)
